src/lib/tls/tls12/tls_channel_impl_12.cpp

```diff
@@ -371,6 +371,9 @@ size_t Channel_Impl_12::from_peer(std::span<const uint8_t> data) {
    } catch(Decoding_Error&) {
       send_fatal_alert(Alert::DecodeError);
       throw;
+   } catch(Invalid_Argument&) {
```

Comment:
One small concern I'd have here is that if we have an uncaught `Invalid_Argument` not due to the X25519/X448 all zeros check, we will send an incorrect and probably quite confusing/misleading alert...

Reply:
Fair. Probably we should aim to not remove those catch-all error matches in general and do the remapping closer to the suspected culprits. For instance, `Invalid_Authentication_Tag` should probably be handled closer to the actual call to the AEAD.